tests/baselines/reference/regularExpressionScanning(target=es5).errors.txt

```diff
@@ -261,6 +265,16 @@ regularExpressionScanning.ts(47,101): error TS1501: This regular expression flag
     	/(hi)\1/,
     	/(hi) (hello) \2/,
     	/\2()(\12)(foo)\1\0[\0\1\01\123](\3\03)\5\005/,
+    	                      ~~
+!!! error TS1487: Octal escape sequences are not allowed. Use the syntax '\x01'.
+    	                        ~~~
+!!! error TS1487: Octal escape sequences are not allowed. Use the syntax '\x01'.
+    	                           ~~~~
+!!! error TS1487: Octal escape sequences are not allowed. Use the syntax '\x53'.
+    	                                   ~~~
+!!! error TS1487: Octal escape sequences are not allowed. Use the syntax '\x03'.
+    	                                         ~~~~
+!!! error TS1487: Octal escape sequences are not allowed. Use the syntax '\x05'.

```

Comment:
I would like to confirm again why decimal escapes (that don’t refer to existent capturing groups) are worth not being errored? Are octal escapes that common?

Reply:
We _should_ report an error for octal escapes. They are permitted in Annex B, but every use I've seen in the wild has actually been a bug, i.e., people mistakenly thinking `/(['"])[^\1]?\1/` matches a quoted string except `[^\1]` is not "every character except the one captured by the backreference `\1`", it's actually "every character except `\x01`".

Reply:
> I mean that outside character classes. 

We should make it an error again in or out of character classes.

> If we are going to emit errors for all octal escapes, [this change](https://github.com/microsoft/TypeScript/pull/58295/files#diff-e19070e6b0ecc4de5f7d4bc6b42a9a71c8f23ce8ce155f7bd7cd48763170e506R3437-R3439), too, should not have been made.

Correct. That PR swung towards being far more permissive while we were discussing a happy medium we should land on. If it not handled in this PR, I was planning to follow up with that once this PR was merged.